source/Octopus.Tentacle/Scripts/ScriptState.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using Octopus.Tentacle.Contracts;
+
+namespace Octopus.Tentacle.Scripts
+{
+    public class ScriptState
+    {
+        public ScriptState(ScriptTicket scriptTicket)
+        {
+            ScriptTicket = scriptTicket;
+        }
+
+        public ScriptTicket ScriptTicket { get; set; }
```

Comment:
Consider making the setters private so there is no way to workaround the logic encapsulated in methods like `Complete`.

Reply:
Good spot 👍